recipe/run_test.py

```diff
@@ -1,8 +1,17 @@
 import sys
-import matplotlib
 
-matplotlib.use('Agg')
+try:
+    # in anticipation of matplotlib not being a core dependency
+    import matplotlib
+    matplotlib.use('Agg')
+except ImportError:
+    #  ModuleNotFoundError when 3.5 gets dropped.
+    pass
 
-import skimage
+import pytest
 
-sys.exit(skimage.test(verbose=True))
+failure_code = pytest.main(['--pyargs',  'skimage'])
```

Comment:
If we are calling is like a CLI, would it make sense to just move this to the `test/commands` section?

ref: https://conda.io/docs/user-guide/tasks/build-packages/define-metadata.html#test-commands

Reply:
I'm not sure since we still need to set the matplotlib backend. Maybe this can be put in a `todo` when don't need the matplotlib lines:

```
try:
    # in anticipation of matplotlib not being a core dependency
    import matplotlib
    matplotlib.use('Agg')
except ImportError:
    #  ModuleNotFoundError when 3.5 gets dropped.
    pass
```

Reply:
We could configure it with the [`MPLBACKEND`]( https://matplotlib.org/faq/environment_variables_faq.html#envvar-MPLBACKEND ) environment variable. This would simplify things a bit. Not pressing though.

Reply:
genius! I'll have to play around with this in a bit.